lib/Sema/MiscDiagnostics.cpp

```diff
@@ -1282,25 +1282,19 @@ static void diagSyntacticUseRestrictions(const Expr *E, const DeclContext *DC,
 
     }
     
-    /// Return true if this is a 'nil' literal.  This looks
-    /// like this if the type is Optional<T>:
-    ///
-    ///   (dot_syntax_call_expr implicit type='Int?'
-    ///     (declref_expr implicit decl=Optional.none)
-    ///     (type_expr type=Int?))
-    ///
-    /// Or like this if it is any other ExpressibleByNilLiteral type:
-    ///
-    ///   (nil_literal_expr)
-    ///
-    bool isTypeCheckedOptionalNil(Expr *E) {
+    /// Returns true if this is a 'nil' literal
+    bool isNilLiteral(Expr *E) {
       if (dyn_cast<NilLiteralExpr>(E)) return true;
-
+      return false;
+    }
+    
+    /// Returns true if this is a expression is a reference to
```

Comment:
Nit: `is a` look extraneous here 
```suggestion
    /// Returns true if this expression is a reference to
```